src/Arcus.Scripting.Management/Scripts/Remove-AzApiManagementSoftDeletedService.ps1

```diff
@@ -0,0 +1,38 @@
+Param(
+    [Parameter(Mandatory = $true)][string] $Name = $(throw "Name of the API Management instance is required"),
+    [Parameter(Mandatory = $false)][string] $SubscriptionId = "",
+    [Parameter(Mandatory = $false)][string] $AccessToken = ""
+)
+
+if($SubscriptionId -eq "" -or $AccessToken -eq ""){
+    # Request accessToken in case the script contains records
+    $token = Get-AzCachedAccessToken
+
+    $AccessToken = $token.AccessToken
+    $SubscriptionId = $token.SubscriptionId
+}
```

Comment:
I was wondering why we did this, or if this comment is actually correct. Seems to me that it should be `...in case the script contains no records.` 😅 But I saw the same thing indeed with Logic Apps. Hmm. 🤔

Reply:
You are right, should be `contains no records`, you want me to change this and in the Logic Apps part as well?

Reply:
Maybe that will clutter things in this PR? You can choose 😄 . I won't mind if we do this in another PR.

Reply:
As it is just a comment, lets do it in a seperate PR later on 👍🏻

Reply:
I will change it for this code and fix the Logic Apps later on in a seperate PR 👍🏻

Reply:
Should be fixed now, see latest commit